src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,131 @@
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
-
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
-
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
-
-	copy(feats);
-}
-
-RandomFourierGaussPreproc::~RandomFourierGaussPreproc() {
-
-	SG_FREE(randomcoeff_multiplicative);
-	SG_FREE(randomcoeff_additive);
+RandomFourierGaussPreproc::~RandomFourierGaussPreproc()
+{
 
 }
 
-EFeatureClass RandomFourierGaussPreproc::get_feature_class() {
+EFeatureClass RandomFourierGaussPreproc::get_feature_class()
+{
 	return C_DENSE;
 }
 
-EFeatureType RandomFourierGaussPreproc::get_feature_type() {
+EFeatureType RandomFourierGaussPreproc::get_feature_type()
+{
 	return F_DREAL;
 }
 
-int32_t RandomFourierGaussPreproc::get_dim_feature_space() const {
-	return ((int32_t) dim_feature_space);
-}
-
-void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_feature_space = dim;
-
-}
-
-int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
-}
-
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
-	}
-	kernelwidth=kernelwidth2;
-}
-
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
-}
-
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
+int32_t RandomFourierGaussPreproc::get_dim_feature_space() const noexcept
+{
+	return m_dims;
 }
 
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
-
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
-
-	return false;
+void RandomFourierGaussPreproc::set_dim_feature_space(int32_t dim)
+{
+	require(m_dims > 0, "Feature space dimension must be greater than zero");
```

Comment:
since it needs to be positive, this should be specified when it is registered. And then this method (and the getter) can also be removed

Reply:
Could you expand on "should be specified when it's registered"?. 
Something like this perhaps?
```c++
fit(Features f, m_kernelwidth,m_dim_output)
```
The above isn't possible because the base class  interface is defined as ```fit(Features f)```
Maybe I'm getting it wrong, haha. Do let me know!

Reply:
no no. You can pass constraints when registering parameters with SG_ADD

Reply:
check `base/constraint.h`

Reply:
KMeansBase.cpp:
```
      SG_ADD(
          &k, "k", "k, the number of clusters",
          ParameterProperties::HYPER | ParameterProperties::CONSTRAIN,
          SG_CONSTRAINT(positive<>()));
```

Reply:
https://github.com/shogun-toolbox/shogun/blob/9b5793b6dff592a9caac3b41f5359838e3ac2ce4/src/shogun/clustering/KMeansBase.cpp#L306-L309

Reply:
this will throw errors when setting parameters with `put`

Reply:
Thanks really!!

Reply:
Ok, how does this correlate to removing getters and setters?. I removed the ```require()``` part

Reply:
remove them

Reply:
I know this may sound pesky but, how would we change m_kernelwidth/m_dim_output without setters ?.

Reply:
with `put` and `get`
If you read one of the examples then you will see how it works. Parameters are registered (`SG_ADD`) so that they can be accessed with those methods.